src/core/ddsi/include/dds/ddsi/ddsi_protocol.h

```diff
@@ -59,6 +59,13 @@ typedef struct {
   uint8_t id[2];
 } ddsi_vendorid_t;
 
+typedef struct {
+  uint8_t major;
+  uint8_t minor;
+  uint8_t release;
+  uint8_t revision;
```

Comment:
Right now it is major, minor, "something" ("patch"?) and I don't know that having four fields won't make life more complicated. What I would find really useful is to also have a commit hash in there (or perhaps just the first few bytes to keep the size in check).